File: src/main/java/net/Chidoziealways/everythingjapanese/potion/ModPotions.java
package net.Chidoziealways.everythingjapanese.potion;

import net.Chidoziealways.everythingjapanese.EverythingJapanese;
import net.Chidoziealways.everythingjapanese.effect.ModEffects;
import net.minecraft.world.effect.MobEffectInstance;
import net.minecraft.world.item.alchemy.Potion;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import net.minecraftforge.registries.RegistryObject;

public class ModPotions {
    public static final DeferredRegister<Potion> POTIONS =
            DeferredRegister.create(ForgeRegistries.POTIONS, EverythingJapanese.MOD_ID);

    public static final RegistryObject<Potion> ADRENALINE_POTION = POTIONS.register("adrenaline_potion",
            () -> new Potion("adrenaline_potion", new MobEffectInstance(ModEffects.ADRENALINE_EFFECT.getHolder().get(), 200, 0)));

    public static void register(IEventBus eventBus){
        POTIONS.register(eventBus);
    }
}

File: src/main/kotlin/net/Chidoziealways/everythingjapanese/potion/ModPotions.kt
package net.Chidoziealways.everythingjapanese.potion

import net.Chidoziealways.everythingjapanese.JAPANESE_MOD_ID
import net.Chidoziealways.everythingjapanese.effect.ModEffects
import net.minecraft.core.registries.Registries
import net.minecraft.world.effect.MobEffectInstance
import net.minecraft.world.item.alchemy.Potion
import net.neoforged.bus.api.IEventBus
import net.neoforged.neoforge.registries.DeferredRegister
import java.util.function.Supplier

object ModPotions {
    val POTIONS = DeferredRegister.create(Registries.POTION, JAPANESE_MOD_ID)

    val ADRENALINE_POTION = POTIONS.register(
        "adrenaline_potion",
        Supplier {
            Potion(
                "adrenaline_potion",
                MobEffectInstance(ModEffects.ADRENALINE_EFFECT.delegate, 200, 0)
            )
        })

    fun register(eventBus: IEventBus) {
        POTIONS.register(eventBus)
    }
}
